library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/HlsPlaylistParser.java

```diff
@@ -300,38 +302,11 @@ private static HlsMasterPlaylist parseMasterPlaylist(LineIterator iterator, Stri
           sessionKeyDrmInitData.add(new DrmInitData(scheme, schemeData));
         }
       } else if (line.startsWith(TAG_STREAM_INF)) {
+        String formatId = Integer.toString(variants.size());
+
         noClosedCaptions |= line.contains(ATTR_CLOSED_CAPTIONS_NONE);
-        int bitrate = parseIntAttr(line, REGEX_BANDWIDTH);
-        String averageBandwidthString =
-            parseOptionalStringAttr(line, REGEX_AVERAGE_BANDWIDTH, variableDefinitions);
-        if (averageBandwidthString != null) {
-          // If available, the average bandwidth attribute is used as the variant's bitrate.
-          bitrate = Integer.parseInt(averageBandwidthString);
-        }
-        String codecs = parseOptionalStringAttr(line, REGEX_CODECS, variableDefinitions);
-        String resolutionString =
-            parseOptionalStringAttr(line, REGEX_RESOLUTION, variableDefinitions);
-        int width;
-        int height;
-        if (resolutionString != null) {
-          String[] widthAndHeight = resolutionString.split("x");
-          width = Integer.parseInt(widthAndHeight[0]);
-          height = Integer.parseInt(widthAndHeight[1]);
-          if (width <= 0 || height <= 0) {
-            // Resolution string is invalid.
-            width = Format.NO_VALUE;
-            height = Format.NO_VALUE;
-          }
-        } else {
-          width = Format.NO_VALUE;
-          height = Format.NO_VALUE;
-        }
-        float frameRate = Format.NO_VALUE;
-        String frameRateString =
-            parseOptionalStringAttr(line, REGEX_FRAME_RATE, variableDefinitions);
-        if (frameRateString != null) {
-          frameRate = Float.parseFloat(frameRateString);
-        }
+        int bitrate = parseStreamBitrate(variableDefinitions, line);
+        Format format = parseVideoContainerFormat(variableDefinitions, line, formatId, bitrate, 0);
         String videoGroupId = parseOptionalStringAttr(line, REGEX_VIDEO, variableDefinitions);
```

Comment:
I refactored to lift methods for the two code blocks to parse out the bitrate and the codecs/frame rate.   As regular Variants (`EXT-X-STREAM-INF`) are nearly identical to I-Frame only variants (`EXT-X-I-FRAME-STREAM-INF`)